nemo/export/utils.py

```diff
@@ -0,0 +1,148 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import logging
+import os
+import pathlib
+import tarfile
+import tempfile
+import typing
+from pathlib import Path
+
+import numpy as np
+import torch
+import yaml
+
+log_format = "%(asctime)s %(name)s [%(levelname)s] %(message)s"
+logging.basicConfig(format=log_format)
+LOGGER = logging.getLogger("NeMo")
+
+# numpy doesn't know bfloat16, define abstract binary type instead
+np_bfloat16 = np.dtype('V2', metadata={"dtype": "bfloat16"})
+
+
+def unpack_nemo_ckpt(
+    nemo_archive_path: typing.Union[str, pathlib.Path], out_dir_path: typing.Union[str, pathlib.Path],
+):
```

Comment:
Please properly document all utility functions (to the best effort)

Reply:
First of all, sorry for dumping all these code in a PR. I know it's a lot but there is really not much I could do. Most of the code came from the TRT-LLM team.

The functions like above are mostly internal functions. The main one that really needs docstrings, comments, etc, is this one https://github.com/oyilmaz-nvidia/NeMo/blob/oss-export-deploy/nemo/export/tensorrt_llm.py and I tried to add comments as much as possible.

Reply:
Moved this utils.py under trt_llm folder since it contains internal functions for trt-llm export.

Reply:
No worries, we've had large PRs before, but we'd like to maintain some level of readable code. We do document utility functions too, but for this case this much is enough